basiclibrary/src/main/java/basiclibrary/Library.java

```diff
@@ -21,10 +21,32 @@ public boolean someLibraryMethod() {
         return numberArray;
     }
 
-//    public boolean containsDuplicates(char[] charArray) {
-//
-//    }
+    public boolean containsDuplicates(char[] charArray) {
+        boolean containDuplicates = false;
+        char compareVariable = ' ';
+        for(int i = 0; i < charArray.length; i++){
+            compareVariable = charArray[i];
+            if(compareVariable == charArray[i]) {
+                containDuplicates = true;
+            } else {
+                containDuplicates = false;
+            }
+        }
+        return containDuplicates;
+    }
+
+    public double calculateAverage(double[] arrayToBeCalculated) {
+        double sum = 0;
+        double average = 0;
 
+        for(double number : arrayToBeCalculated) {
+            sum += number;
+        }
+        average = sum / arrayToBeCalculated.length();
```

Comment:
Code line 45: average = sum / arrayToBeCalculated.length; 
You added () to .length and it is breaking the code build. If you remove them the code will compile and your first test passes.